platform/linux-generic/odp_queue_basic.c

```diff
@@ -688,6 +735,11 @@ int _odp_event_aggr_enq(queue_entry_t *aggr_queue, _odp_event_hdr_t *event_hdr[]
 
 	_ODP_ASSERT(aggr_queue->type == ODP_QUEUE_TYPE_AGGR);
 
+	/* Packet input may later drop a vector if the base queue is full, so enqueue events to the
+	 * base queue directly. */
+	if (odp_unlikely(base_queue->pktin.pktio != ODP_PKTIO_INVALID))
+		return base_queue->enqueue_multi(base_queue->handle, event_hdr, num);
```

Comment:
I suppose this special case here would not be necessary here if the enqueue functions for aggregators of pktin queues would be the enqueue function of the base queue. I do not immediately see why not. But ok to leave that as a future improvement.